File: src/main/java/com/imagegame/service/LeaderboardService.java
package com.imagegame.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.imagegame.domain.Leaderboard;
import com.imagegame.dto.LeaderboardDTO;
import com.imagegame.dto.LeaderboardSearchDTO;
import com.imagegame.dto.LeaderboardPageDTO;
import com.imagegame.dto.LeaderboardConvertCriteriaDTO;
import com.imagegame.service.GenericService;
import com.imagegame.dto.common.RequestDTO;
import com.imagegame.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface LeaderboardService extends GenericService<Leaderboard, Integer> {

	List<Leaderboard> findAll();

	ResultDTO addLeaderboard(LeaderboardDTO leaderboardDTO, RequestDTO requestDTO);

	ResultDTO updateLeaderboard(LeaderboardDTO leaderboardDTO, RequestDTO requestDTO);

    Page<Leaderboard> getAllLeaderboards(Pageable pageable);

    Page<Leaderboard> getAllLeaderboards(Specification<Leaderboard> spec, Pageable pageable);

	ResponseEntity<LeaderboardPageDTO> getLeaderboards(LeaderboardSearchDTO leaderboardSearchDTO);
	
	List<LeaderboardDTO> convertLeaderboardsToLeaderboardDTOs(List<Leaderboard> leaderboards, LeaderboardConvertCriteriaDTO convertCriteria);

	LeaderboardDTO getLeaderboardDTOById(Integer leaderboardId);







}

File: imagegame_ui/src/services/LeaderboardService.js
import http from "../http-common"; 

class LeaderboardService {
  getAllLeaderboards(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/leaderboard/leaderboards`, searchDTO);
  }

  get(leaderboardId) {
    return this.getRequest(`/leaderboard/${leaderboardId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/leaderboard?field=${matchData}`, null);
  }

  addLeaderboard(data) {
    return http.post("/leaderboard/addLeaderboard", data);
  }

  update(data) {
  	return http.post("/leaderboard/updateLeaderboard", data);
  }
  
  uploadImage(data,leaderboardId) {
  	return http.postForm("/leaderboard/uploadImage/"+leaderboardId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new LeaderboardService();
